OtripleS.Portal.Web/Services/Teachers/TeacherService.Exceptions.cs

```diff
@@ -0,0 +1,85 @@
+﻿// ---------------------------------------------------------------
+// Copyright (c) Coalition of the Good-Hearted Engineers
+// FREE TO USE AS LONG AS SOFTWARE FUNDS ARE DONATED TO THE POOR
+// ---------------------------------------------------------------
+
+using System;
+using System.Collections.Generic;
+using System.Net.Http;
+using System.Threading.Tasks;
+using OtripleS.Portal.Web.Models.Teachers;
+using OtripleS.Portal.Web.Models.Teachers.Exceptions;
+using RESTFulSense.Exceptions;
+
+namespace OtripleS.Portal.Web.Services.Teachers
+{
+    public partial class TeacherService
+    {
+        private delegate ValueTask<IReadOnlyList<Teacher>> ReturningTeacherFunction();
+
+        private async ValueTask<IReadOnlyList<Teacher>> TryCatch(ReturningTeacherFunction returningTeacherFunction)
+        {
+            try
+            {
+                return await returningTeacherFunction();
+            }
+            catch (HttpRequestException httpRequestException)
+            {
+                throw CreateAndLogCriticalDependencyException(httpRequestException);
+            }
+            catch (HttpResponseUrlNotFoundException httpResponseUrlNotFoundException)
+            {
+                throw CreateAndLogCriticalDependencyException(httpResponseUrlNotFoundException);
+            }
+            catch (HttpResponseUnauthorizedException httpResponseUnauthorizedException)
+            {
+                throw CreateAndLogCriticalDependencyException(httpResponseUnauthorizedException);
+            }
+            catch (HttpResponseInternalServerErrorException httpResponseInternalServerException)
```

Comment:
@ShriHumrudha what do you mean? This is a dependency exception. In fact I don't think we need to handle `HttpResponseInternalServerErrorException` at all. Because it should be already covered under the `HttpResponseException` portion.